rca/api_content/content.py

```diff
@@ -59,8 +60,12 @@ def parse_items_to_list(data, type):
                 + "?fields=_,first_published_at,social_image,body"
             )
         try:
```

Comment:
This `try/except` pattern is repeated a few times, could it be wrapped in a function?

Reply:
Thanks Rich, good idea. I've attempted this here https://github.com/torchbox/rca-wagtail-2019/pull/313/commits/02e690aa81a9c9317b9b473a0339a359cf03d57a.

I'm not 100% on this, mainly because if the execution makes it to here https://github.com/torchbox/rca-wagtail-2019/pull/313/commits/02e690aa81a9c9317b9b473a0339a359cf03d57a#diff-3a8951d04a021439a3fbc9866c9a2d16R155 then there will be no response defined; but I guess before that the exception would be raised? Should I be checking for the response at this point?

Reply:
Also, I've changed the timeout to be 5 seconds which is a bit more reasonable

Reply:
RE the `try_and_fetch()` function, another repeated line is `data = response.json()`, which suggests that what your code really wants from the function is appropriate data, rather than a response that it then needs to deal with (and which it doesn't need/want to know about).

You could rename the function to be something like `fetch_data(url)` to encapsulate all the `requests` boilerplate, and maybe return `None` if it can't fetch it so your code can check.

Reply:
I've updated the code following your suggestions.
I've also tested this by replacing https://github.com/torchbox/rca-wagtail-2019/pull/313/files#diff-3a8951d04a021439a3fbc9866c9a2d16R48 with `data = []` to test the page still loads with no data.

Ideally, I need some tests for this, which I will add as a backlog ticket.